src/main/java/com/purbon/kafka/topology/TopicManager.java

```diff
@@ -77,20 +72,43 @@ public void apply(Topology topology, ExecutionPlan plan) throws IOException {
     }
   }
 
+  private Map<String, Topic> parseMapOfTopics(Topology topology) {
+    return topology.getProjects().stream()
+        .flatMap(project -> project.getTopics().stream())
+        .filter(this::matchesPrefixList)
+        .collect(Collectors.toMap(Topic::toString, topic -> topic));
+  }
+
   private boolean isAnInternalTopics(String topic) {
     return internalTopicPrefixes.stream().anyMatch(topic::startsWith);
   }
 
   private Set<String> loadActualClusterStateIfAvailable(ExecutionPlan plan) throws IOException {
     Set<String> listOfTopics =
-        config.fetchStateFromTheCluster() ? adminClient.listApplicationTopics() : plan.getTopics();
+        config.fetchTopicStateFromTheCluster()
```

Comment:
Hi, I have a philosophical question.

With the introduction of the fetchStateFromCluster global config variable to control when the tool is fetching data from the cluster, or from the local plan, for backwards compatibility we still have the specific one for topics. I wonder this will be confusing for people.

Do you think we can simply have a global fetchStateFromTheCluster and release a 2.0 version? this would be less confusing I think. What do you think?